src/main/java/eu/tsystems/mms/tic/testerra/plugins/xray/synchronize/AbstractXrayResultsSynchronizer.java

```diff
@@ -221,22 +221,29 @@ private synchronized void flushSyncQueue() {
         boolean areNewTestsToImport = testSetSyncQueue.stream().anyMatch(xrayTestSetIssue -> {
             return xrayTestSetIssue.getTestKeys().stream().anyMatch(key -> key.contains(XrayUtils.PREFIX_NEW_ISSUE));
         });
-        if (areNewTestsToImport) {
-            xrayTestExecutionImport.getResultTestIssueImport().getSuccess().forEach(jiraIssueReference -> {
-                try {
-                    // Replace the temporary key with real Jira key from the result 'xrayTestExecutionImport'
-                    JiraIssue issue = xrayUtils.getIssue(jiraIssueReference.getKey());
-                    testSetSyncQueue.forEach(xrayTestSetIssue -> {
-                        Optional<String> findKey = xrayTestSetIssue.getTestKeys().stream().filter(key -> key.contains(issue.getSummary())).findFirst();
-                        if (findKey.isPresent()) {
-                            xrayTestSetIssue.getTestKeys().removeIf(key -> key.contains(issue.getSummary()));
-                            xrayTestSetIssue.getTestKeys().add(issue.getKey());
-                        }
-                    });
-                } catch (IOException e) {
-                    log().error(String.format("Unable to read %s", IssueType.TestExecution), e);
-                }
-            });
+        if (areNewTestsToImport && xrayTestExecutionImport.getResultTestIssueImport() != null) {
+            if (xrayTestExecutionImport.getResultTestIssueImport().getError().size() > 0) {
+                log().error("Error at syncing with Jira");
+                xrayTestExecutionImport.getResultTestIssueImport().getError()
+                        .forEach(error -> log().error("{} - {}: {}", error.getProject(), error.getSummary(), error.getMessages().toString()));
```

Comment:
It would be better to concatenate these into _one_ log statement together with the very plain "Error at syncing with Jira", such that said statement cannot be separated e.g. if several loggers write into the same file. As-is the individual statements may be scattered arbitrarily far in a sufficiently degenerate logger setup.